File: app/src/main/java/de/bitb/astroskop/utils/BluetoothUtils.java
package de.bitb.astroskop.utils;


import android.bluetooth.BluetoothAdapter;

import de.bitb.astroskop.exceptions.TechnologyDisabledException;
import de.bitb.astroskop.exceptions.TechnologyNotExistingException;


public class BluetoothUtils {

    public void checkForBluetooth() throws TechnologyNotExistingException, TechnologyDisabledException {
        BluetoothAdapter adapter = BluetoothAdapter.getDefaultAdapter();

        if (adapter == null) {
            throw new TechnologyNotExistingException();
        } else if (!adapter.isEnabled()) {
            throw new TechnologyDisabledException();
        }
    }
}

File: app/src/main/java/de/bitb/astroskop/utils/BluetoothUtils.kt
package de.bitb.astroskop.utils


import android.bluetooth.BluetoothAdapter

import de.bitb.astroskop.exceptions.TechnologyDisabledException
import de.bitb.astroskop.exceptions.TechnologyNotExistingException


class BluetoothUtils {

    @Throws(TechnologyNotExistingException::class, TechnologyDisabledException::class)
    fun checkForBluetooth() {
        val adapter = BluetoothAdapter.getDefaultAdapter()

        if (adapter == null) {
            throw TechnologyNotExistingException()
        } else if (!adapter.isEnabled) {
            throw TechnologyDisabledException()
        }
    }
}
